astropy/nddata/ccddata.py

```diff
@@ -0,0 +1,568 @@
+            header['bunit'] = self.unit.to_string()
+        if self.wcs:
+            # Simply extending the FITS header with the WCS can lead to
+            # duplicates of the WCS keywords; iterating over the WCS
+            # header should be safer.
+            #
+            # Turns out if I had read the io.fits.Header.extend docs more
+            # carefully, I would have realized that the keywords exist to
+            # avoid duplicates and preserve, as much as possible, the
+            # structure of the commentary cards.
+            #
+            # Note that until astropy/astropy#3967 is closed, the extend
+            # will fail if there are comment cards in the WCS header but
+            # not header.
+            wcs_header = self.wcs.to_header(relax=wcs_relax)
+            header.extend(wcs_header, useblanks=False, update=True)
+        hdus = [fits.PrimaryHDU(self.data, header)]
+
+        if hdu_mask and self.mask is not None:
+            # Always assuming that the mask is a np.ndarray (check that it has
+            # a 'shape').
+            if not hasattr(self.mask, 'shape'):
+                raise ValueError('only a numpy.ndarray mask can be saved.')
+
+            # Convert boolean mask to uint since io.fits cannot handle bool.
+            hduMask = fits.ImageHDU(self.mask.astype(np.uint8), name=hdu_mask)
+            hdus.append(hduMask)
+
+        if hdu_uncertainty and self.uncertainty is not None:
+            # We need to save some kind of information which uncertainty was
+            # used so that loading the HDUList can infer the uncertainty type.
+            # No idea how this can be done so only allow StdDevUncertainty.
+            if self.uncertainty.__class__.__name__ != 'StdDevUncertainty':
+                raise ValueError('only StdDevUncertainty can be saved.')
+
+            # Assuming uncertainty is an StdDevUncertainty save just the array
+            # this might be problematic if the Uncertainty has a unit differing
+            # from the data so abort for different units. This is important for
+            # astropy > 1.2
+            if (hasattr(self.uncertainty, 'unit') and
+                    self.uncertainty.unit is not None and
+                    self.uncertainty.unit != self.unit):
+                raise ValueError('saving uncertainties with a unit differing'
+                                 'from the data unit is not supported.')
+
+            hduUncert = fits.ImageHDU(self.uncertainty.array,
+                                      name=hdu_uncertainty)
+            hdus.append(hduUncert)
+
+        if hdu_flags and self.flags:
+            raise NotImplementedError('adding the flags to a HDU is not '
+                                      'supported at this time.')
+
+        hdulist = fits.HDUList(hdus)
+
+        return hdulist
+
+    def copy(self):
+        """
+        Return a copy of the CCDData object.
+        """
+        try:
```

Comment:
I don't think there's any need for the `except` block. That was only a workaround because astropy < 1.2 did not support `copy`.